src/clj/battlebots/utils/arena.clj

```diff
@@ -0,0 +1,177 @@
+(ns battlebots.utils.arena
+  (:require [battlebots.constants.arena :refer [arena-key]]))
+
+(defn empty-arena
+  "returns an empty arena"
+  [dimx dimy]
+  (vec (repeat dimx (vec (repeat dimy (:open arena-key))))))
+
+(defn get-item
+  "gets an item based off of given coords"
+  [coords arena]
+  (get-in arena coords))
+
+(defn get-arena-dimensions
+  "returns the dimensions of a given arena (NOTE: Not 0 based)"
+  [arena]
+  (let [x (count arena)
+        y ((comp count first) arena)]
+    [x y]))
+
+(defn get-arena-row-cell-length
+  "Similar to arena/get-arena-dimensions except that it is zero based"
+  [arena]
+  (map dec (get-arena-dimensions arena)))
+
+(defn wrap-coords
+  "wraps the coords around to the other side of the arena"
+  [[c-x c-y] [d-x d-y]]
+  (let [x (cond
+            (< c-x 0) (if (> (Math/abs c-x) d-x)
+                        (- d-x (mod (Math/abs c-x) d-x))
+                        (+ d-x c-x))
+            (> c-x d-x) (mod c-x d-x)
+           :else c-x)
+        y (cond
+            (< c-y 0) (if (> (Math/abs c-y) d-y)
+                        (- d-y (mod (Math/abs c-y) d-y))
+                        (+ d-y c-y))
+            (> c-y d-y) (mod c-y d-y)
+            :else c-y)]
+    [x y]))
+
+(defn incx
+  [x]
+  (fn [v] (+ x v)))
+
+(defn adjust-coords
+  "Returns a new set of coords based off of an applied direction.
+
+  0 1 2
+  7   3
+  6 5 4
+
+  "
+  ([coords direction dimensions] (adjust-coords coords direction dimensions 1))
```

Comment:
This is really cool, I always forget about this option for setting default values